agimus_demos_common/agimus_demos_common/wait_for_non_zero_joints_node.py

```diff
@@ -82,29 +70,21 @@ def _joint_states_cb(self, msg: JointState) -> None:
         self.get_logger().info("Joint state message received.", once=True)
 
         if sum([abs(p) for p in msg.position]) > self._joints_sum_threshold:
-            self._exit_code = 0
-            raise SystemExit()
+            self.get_logger().info("Received non-zero joint position.")
+            sys.exit(0)
 
         if (self.get_clock().now() - self._init_time) > self._timeout:
-            self.get_logger().info(
+            self.get_logger().error(
                 "Failed to receive non-zero joint "
                 + "position before timeout was reached."
             )
-            self._exit_code = 1
-            raise SystemExit()
+            sys.exit(1)
 
 
 def main(args=None):
     rclpy.init(args=args)
-    ret = 1
     wait_for_non_zero_joints_node = WaitForNonZeroJoints()
-    try:
-        rclpy.spin(wait_for_non_zero_joints_node)
-    except SystemExit:
-        wait_for_non_zero_joints_node.destroy_node()
-        ret = wait_for_non_zero_joints_node.exit_code
-        rclpy.try_shutdown()
-    sys.exit(ret)
+    rclpy.spin(wait_for_non_zero_joints_node)
```

Comment:
Can there be some artifacts remaining if the node is not destroyed?
Maybe we can use:
```python
try:
    rclpy.spin(wait_for_non_zero_joints_node)
finally:
    wait_for_non_zero_joints_node.destroy_node()
```

Reply:
mmmh maybe I can have a second look.

Reply:
Done